libpod/image/image.go

```diff
@@ -133,7 +133,7 @@ func (ir *Runtime) NewFromLocal(name string) (*Image, error) {
 
 // New creates a new image object where the image could be local
 // or remote
-func (ir *Runtime) New(ctx context.Context, name, signaturePolicyPath, authfile string, writer io.Writer, dockeroptions *DockerRegistryOptions, signingoptions SigningOptions, forcePull bool, label *string) (*Image, error) {
+func (ir *Runtime) New(ctx context.Context, name, signaturePolicyPath, authfile string, writer io.Writer, dockeroptions *DockerRegistryOptions, signingoptions SigningOptions, forcePull bool, label *string, update bool) (*Image, error) {
```

Comment:
We really need to convert this to accept a struct at some point, the signature has gotten well and truly out of hand

Reply:
+1

Reply:
made this comment earlier ... you are hitting all the instances anyways, lets do it now.

Reply:
Should the `forcePull` and `update` bools also be named fields of the struct instead of unnamed and somewhat likely-to-be-confused parameters to the function?

Reply:
I guess I see no problem with that @mtrmac  other than them being useless args in pullImageFromHeuristicSource* but if that isn't deemed a problem to anyone else then I'll fix.

Reply:
See the rest of that review first… this `update`/`pullAlways` implementation might end up different (or not; I don’t know at all yet).

To get exactly the current behavior with parameters in structs, we could also
```go
type PullOptions struct {
   // the current members
}
type NewImageConfig struct {
  PullOptions
  PullAlways, Update bool
  // whatever else makes sense
}
```
I’m not 100% sure whether that‘s more or less readable.